service/indexer.go

```diff
@@ -14,13 +19,30 @@ func (vs *VocdoniService) VochainIndexer() error {
 	vs.Indexer, err = indexer.New(vs.App, indexer.Options{
 		DataDir:           filepath.Join(vs.Config.DataDir, "indexer"),
 		IgnoreLiveResults: vs.Config.Indexer.IgnoreLiveResults,
+		// During StateSync, IndexerDB will be restored, so enable ExpectBackupRestore in that csae
+		ExpectBackupRestore: vs.Config.StateSyncEnabled,
 	})
 	if err != nil {
 		return err
 	}
 	// launch the indexer after sync routine (executed when the blockchain is ready)
 	go vs.Indexer.AfterSyncBootstrap(false)
 
+	snapshot.SetFnImportIndexer(vs.Indexer.RestoreBackup)
+
+	snapshot.SetFnExportIndexer(func(w io.Writer) error {
+		exportCtx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		defer cancel()
+		data, err := vs.Indexer.ExportBackupAsBytes(exportCtx)
+		if err != nil {
+			return fmt.Errorf("error reading indexer backup: %w", err)
+		}
+		if _, err := w.Write(data); err != nil {
+			return fmt.Errorf("error writing data: %w", err)
+		}
+		return nil
+	})
```

Comment:
if the import and export funcs take reader and writer, why do the indexer funcs take reader and []byte? the inconsistency feels odd.

FWIW I wrote the indexer methods to use filenames on purpose, because sqlite can only create and restore backups as files on disk. If we need the glue code to use temporary files because you want to use e.g. []byte or Reader/Writer, I'd personally put those here.

Reply:
you're totally right. left your methods untouched and moved the glue into service/indexer.go like you suggested

i did add the new method ExportBackupAsBytes in indexer package since that is reused by an api endpoint